terraform-incubator/people-depot/project/main.tf

```diff
@@ -64,6 +64,15 @@ module "people_depot" {
   root_db_password = var.root_db_password
 }
 
+module "cognito" {
+  source = "../../../terraform-modules/cognito"
+
+  region         = "us-west-2"
+  user_pool_name = "people-depot-user-pool"
+  client_name    = "people-depot-client"
```

Comment:
hey @fyliu - apologies for the late response here...i just noticed these comments yesterday. 

the modules are basically the recipes that any project can use to build what they need. the variables can have default values but those can be overridden if you pass values when calling the module. so, files in /cognito define the resources (main.tf), along w/ their variables and outputs

the main.tf file inside of people-depot/project calls that module and passes the client info (or whatever other configs you want to pass it)

for adding multiple clients, you'd call the module multiple times like so:

```
module "cognito_1" {
  source = "../../../terraform-modules/cognito"

  region         = "us-west-2"
  user_pool_name = "x-user-pool"
  client_name    = "x-client"
}
module "cognito_2" {
  source = "../../../terraform-modules/cognito"

  region         = "us-west-2"
  user_pool_name = "x-user-pool"
  client_name    = "x-client"
}
```
if the configs need to change (like generate_secret = true) - then we can designate those as variables when creating the module and then pass the value when calling the module in project/main.tf

Reply:
also, each project would use the module inside their project directory's main.tf file. if it's a shared resource, we can move it to a shared directory...but it sounded like this would be project specific